Sources/MapboxNavigation/NavigationMapView+BuildingHighlighting.swift

```diff
@@ -15,13 +15,122 @@ extension NavigationMapView {
      - returns: Bool indicating if number of buildings found equals number of coordinates supplied.
      */
     @discardableResult public func highlightBuildings(at coordinates: [CLLocationCoordinate2D], in3D extrudesBuildings: Bool = true) -> Bool {
-        return false
+        let _ = coordinates.map({buildingIdentifier(at: $0, in3D: extrudesBuildings) })
+        DispatchQueue.main.asyncAfter(deadline: .now() + handleAsyncTime, execute: {self.addBuildingsLayer(with: self.foundBuildingIds, in3D: extrudesBuildings)})
```

Comment:
Why do you need to run this command asynchronously? Should be similar to previous implementation (besides completion callback).

Reply:
The previous completion callback in 1.x is not working in the Carbon due to the `mapView.visibleFeatures(at: styleLayers:,completion:` having an escaping closure now. The asynchronous completion result would cause the buildingIds delayed with an empty set passed to `func addBuildingsLayer(with)`

Reply:
In case if `buildingIdentifier` runs async operation it should provide callback as well. `highlightBuildings()` will also have to be updated to provide callback instead of returning boolean.

Reply:
Thanks for the suggestion, I've added the completion and call back for the `func highlightBuildings `.